modules/cloudwatch_metrics/main.tf

```diff
@@ -0,0 +1,58 @@
+locals {
+  role_name     = regex(".*role/(?P<role_name>.*)$", var.lambda.lambda_function.role)["role_name"]
+  function_name = regex(".*:function:(?P<function_name>.*)$", var.lambda.lambda_function.arn)["function_name"]
```

Comment:
Presumably if `var.lambda.lambda_function` is an `aws_lambda` object you can just access the add the function name to the input object so that you can access it here? Rather than back it out of the ARN?

Reply:
Yes, although I was trying to be as axiomatic as possible and allow satisfying the interface through other means. I've replaced this dubious regex with a combination of data.aws_arn and `trimprefix` to make it less obtuse.